cmd/config-brancher/README.md

```diff
@@ -0,0 +1,73 @@
+# config-brancher
+
+This tool is intended to make the process of branching and duplicating configuration for the CI Operator easy across
+many repositories.
+
+## What it does
+
+The tool has two main modes of operations. The first mode is mirroring the CI Operator configuration from the branch
+where the active development occurs (such as `main`) to future release branches.
+
+In the second mode, the tool is used to "bump" the configuration file versions, retargeting the configuration for
+the current development branch to the next version during the code freeze.
+
+In both modes, it also enforces the basic promotion destination consistency.
+
+## Why it exists
+
+The tool allows the users to only maintain a single CI Operator configuration file: the one for main development branch.
+The changes done in this file will then be automatically propagated to future release branches, reducing the surface for
+omissions and mistakes.
+
+In the bump mode, the tool allows Test Platform to automatically perform the CI configuration changes on code freeze (
+a.k.a "branch cut").
+
+## How it works
+
+The tool iterates over all ci-operator config files in openshift/release, and it selects a set of repositories to
+operate on by looking at which are actively promoting images into a specific OpenShift release, provided
+by `--current-release`. Branches of repos that actively promote to this release are considered to be the dev branches.
+
+Once the tool selects the set of configurations to operate on, it does one of two actions: Mirror the configuration out,
+or bump the configuration files.
+
+### Mirroring configuration
+
+In the mirror configuration mode, it copies the development branch config to all branches for the provided
+`--future-release` values, not changing the configuration for the dev branch and making sure that the release branch for
+the version that matches that in the dev branch has a disabled promotion stanza to ensure only one branch feeds a
+release ImageStream.
```

Comment:
This was a bit hard to follow, how about:

```suggestion
In the mirror configuration mode, it copies the development branch configuration to all branches for the provided
`--future-release` values. The development branch version is not changed. Since this results in both the development
branch and one of the release branches promoting to the same release ImageStream, promotion is disabled in the release
branch for the version which matches that in the promotion stanza of the development branch. This ensures only one
branch feeds that release ImageStream.
```

Reply:
Like that!